sw/device/silicon_creator/rom_ext/doc/ownership.md

```diff
@@ -0,0 +1,542 @@
+    uint32_t erase_previous;        // Erase the previous owner's flash (hardened_bool_t).
+    uint32_t reserved[33];          // Reserved for future use.
+    uint64_t nonce;                 // The current ownership nonce.
+    uint32_t signature[16];         // Signature over [primary_bl0_slot..nonce].
+};
+```
+See the definition in [boot_svc_ownership_activate.h](../../lib/boot_svc/boot_svc_ownership_activate.h).
+
+A valid ownership activate request will:
+- Rotate the ROM\_EXT nonce value.
+- Update the ownership state to `LockedOwner`.
+- Set the primary BL0 slot.
+- Erase the previous owner's flash slot (ie: the non-primary slot).
+- Copy the owner configuration from owner page 1 into owner page 0.
+
+## Owner Configuration
+
+### Contents
+
+The owner configuration contains the following items:
+- Owner Key:  This key is the owner's identity and is used to endorse the keys used for application verification and future owner configuration updates or transfer request.
+- Activation Key:  This key is used to authenticate an `OwnershipActivate` command.
+- Unlock Key:  This key is used to authenticate an `OwnershipUnlock` command.
+- Owner Application Keys:  These keys are used to authenticate the owner's firmware payload.
+   The keys also carry some additional metadata used to configure the OpenTitan key manager.
+- EFLASH configuration.
+  The EFLASH configuration sets the desired properties on ranges of flash data pages (such as access permissions and error correction properties).
+- EFLASH-INFO configuration: The EFLASH-INFO configuration sets the desired properties on flash info pages.
+  Some info pages are reserved for use by the `silicon_creator` and have restricted access.
+  The remaining info pages are available to the owner and the owner can set specific access permissions.
+- Miscellaneous settings:
+   - SRAM execution: depending on the application, the owner may want to allow or forbid code execution from SRAM.
+   - Rescue configuration: The ROM\_EXT supports a rescue protocol for recovering from an internal flash corruption event and querying data about the chip.
+     The rescue configuration allows the owner to control interactions with the rescue protocol.
+
+Because the Owner Configuration is composed of several sections of varying size and count (e.g. perhaps several different keys), the owner configuration utilizes tag-length-value (TLV) encoded structures.
+
+```c
+struct TLVHeader {
+    uint32_t tag;
+    uint32_t length;
+};
+```
+See the definition in [datatypes.h](../../lib/ownership/datatypes.h).
+
+### Structures
+
+#### Owner Configuration
+
+The Owner Configuration struct carries the required owner key materials and validation information and is a container for all of the TLV structs specifying the configuration.
+- The `owner_key` Key is an ECDSA P-256 key used to authenticate the owner configuration and endorse the other keys.
```

Comment:
nit: other places in this document don't specify P-256, and by my reading the scheme is not actually sensitive to which signature scheme we use (or even really ECDSA -- we just need some signature mechanism -- but I suspect that might cross the line into being a little too abstract).
```suggestion
- The `owner_key` Key is an ECDSA key used to authenticate the owner configuration and endorse the other keys.
```

Reply:
For ECDSA, P-256 is an assumption.  We could expand the sizes of the key and signature structs to accommodate P-384 as well.  Should we?

Reply:
I don't think there's a need to expand the sizes right now, since we have no plans to move to P-384.

Reply:
Created #22566 to discuss how to handle any non-P256 keys and signatures for ownership transfer.